lib/sensors/mux.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "sensors.hpp"
+
+#include <cstdint>
+#include <cstdio>
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+
+enum class Mode { kWrite = 0, kRead };
+enum class MuxWriteResult { kFailure = 0, kSuccess };
+
+template<class T>
+class Mux {
+ public:
+  Mux(hyped::io::I2c &i2c,
+      const uint8_t mux_address,
+      std::vector<std::unique_ptr<II2cSensor<T>>> sensors,
+      hyped::core::ILogger &log);
+  ~Mux();
+
+  bool selectMode(Mode io_mode);
```

Comment:
const

Reply:
also: return type bool?

Reply:
Wanted to call selectMode in readAll and write before anything actually happens in those functions, but if selectMode fails for whatever reason (since it also involves writing to the I2C mux) -> wanted to instantly return a `std::nulopt` or `kFailure`

Reply:
make it a success or failure return type then; not `bool`